Content.Server/CartridgeLoader/Cartridges/MessagesCartridgeSystem.cs

```diff
@@ -0,0 +1,310 @@
+            messageData.SenderId = component.UserUid;
+            messageData.ReceiverId = component.ChatUid;
+            messageData.Content = messageEvent.Parameter;
+            messageData.Time = _gameTiming.CurTime.Subtract(_gameTicker.RoundStartTimeSpan);
+            component.MessagesQueue.Add(messageData);
+            Update(uid, component);
+        }
+        else
+        {
+            if (messageEvent.Action == MessagesUiAction.ChangeChat)
+                component.ChatUid = messageEvent.Parameter;
+        }
+
+        UpdateUiState(uid, GetEntity(args.LoaderUid), component);
+    }
+
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var cartQuery = EntityQueryEnumerator<MessagesCartridgeComponent>();
+        while (cartQuery.MoveNext(out var uid, out var messagesCartridge))
+        {
+            if (messagesCartridge.NextUpdate > _gameTiming.CurTime)
+                continue;
+            messagesCartridge.NextUpdate = _gameTiming.CurTime + messagesCartridge.UpdateDelay;
+
+            Update(uid, messagesCartridge);
+        }
+    }
+
+    public void Update(EntityUid uid, MessagesCartridgeComponent component)
+    {
+        var mapId = Transform(uid).MapID;
+
+        if (component.UserName == null)
+        {
+            if ((component.ConnectedId != null) && (TryComp(component.ConnectedId, out IdCardComponent? idCardComponent)))
+            {
+                if (idCardComponent.FullName != "")
+                {
+                    component.UserName = idCardComponent.FullName+"("+idCardComponent.JobTitle+")";
+                }
+                else
+                {
+                    component.UserName = "Unknown "+"("+idCardComponent.JobTitle+")";
+                }
+            }
+            if (component.UserUid != null && component.UserName != null)
+                component.NameDict[component.UserUid] = component.UserName;
+        }
+
+        //_sawmill.Debug("Updating cart");
+        if ((component.MessagesQueue.Count > 0) || component.DeadConnection)
+        {
+            if (!(GetActiveServer(component, mapId) is var (serverUid, messageServer)))
+            {
+                component.DeadConnection=true;
+                return;
+            }
+            else
+            {
+                if (component.DeadConnection)
+                    PullFromServer(uid, component, messageServer);
+                component.DeadConnection=false;
+            }
+
+
+            var tempMessageQueue = new List<MessagesMessageData>(component.MessagesQueue);
+            foreach (var message in tempMessageQueue)
+            {
+                _messagesServerSystem.PdaToServerMessage(serverUid, messageServer,message);
+                component.MessagesQueue.Remove(message);
+                component.Messages.Add(message);
+                UpdateUiState(uid, Transform(uid).ParentUid, component);
+            }
+        }
+    }
+
```

Comment:
1. Don't use string concat everywhere as mentioned above.
2. Run a formatter over this.
3. Don't just hardcode in getting the username every tick.
4. This isn't localised.
5. If you need a cartridge to poll add a dedicated component instead of having hundreds of cartridges just polling for servers.